File: app/src/main/java/com/softaai/designpatterns/java/structural/flyweight/Circle.java
package com.softaai.designpatterns.java.structural.flyweight;

/**
 * Created by Amol Pawar on 04-04-2019.
 * softAai Apps
 */
public class Circle implements Shape {

    private String color;
    private int x;
    private int y;
    private int radius;

    public Circle(String color){
        this.color = color;
    }

    public void setX(int x){
        this.x = x;
    }

    public void setY(int y){
        this.y = y;
    }

    public void setRadius(int radius){
        this.radius = radius;
    }

    @Override
    public void draw() {
        System.out.println("Circle : Draw() [Color : " + color + ", x : " + x + ", y : " + y + ", radius : " + radius);
    }
}

File: app/src/main/java/com/softaai/designpatterns/kotlin/structural/flyweight/Circle.kt
package com.softaai.designpatterns.kotlin.structural.flyweight


/**
 * Created by Amol Pawar on 04-04-2019.
 * softAai Apps
 */
class Circle(color:String) : Shape{
    private var color : String
    private var x : Int = 0
    private var y = 0
    private var radius = 0

    init {
        this.color = color
    }

    fun setX(x : Int){
        this.x = x
    }

    fun setY(y : Int){
        this.y = y
    }

    fun setRadius(radius : Int){
        this.radius = radius
    }

    override fun draw() {
        println("Circle : Draw() [Color : " + color + ", x : " + x +  ", y : " + y + ", radius : " + radius + " ]")
    }
}